src/pages/test-results-search/test-results-search.ts

```diff
@@ -78,4 +81,22 @@ export class TestResultsSearchPage extends BasePageComponent {
     }
   }
 
+  advancedSearch(advancedSearchParams: AdvancedSearchParams): void {
+    this.showAdvancedSearchSpinner = true;
+    this.searchProvider.advancedSearch(advancedSearchParams)
+      .pipe(
+        tap(data => console.log('Advanced Search', JSON.stringify(data))),
+        tap(() => this.hasSearched = true),
+        map((result) => {
+          this.searchResults = result;
+          this.showAdvancedSearchSpinner = false;
+        }),
+        catchError((err) => {
+          this.showAdvancedSearchSpinner = false;
+          return of(console.log('ERROR', JSON.stringify(err)));
+        }),
+      )
+      .subscribe();
```

Comment:
Can we use the `| async` pipe in the template to render this? That will also handle unsubscribing

Reply:
Would be good. But let's keep that as an improvement on top of this.